cpp/ql/src/semmle/code/cpp/dataflow/internal/AddressFlow.qll

```diff
@@ -92,6 +103,17 @@ private predicate pointerToPointerStep(Expr pointerIn, Expr pointerOut) {
 
 private predicate lvalueToReferenceStep(Expr lvalueIn, Expr referenceOut) {
   lvalueIn.getConversion() = referenceOut.(ReferenceToExpr)
+  or
+  // C++ only
+  exists(Call c |
+    referenceOut = c and
+    c.getTarget().hasName(["operator*", "operator++"]) and
```

Comment:
I can see that `operator++` is an `lvalueToReferenceStep`, assuming it returns a reference to `*this`, but I don't see how `operator*` can be thought of as being such a step. It should effectively be equivalent to `ReferenceToExpr`. Did you mean to type `operator--` instead?